R/label-number-si.R

```diff
@@ -17,19 +16,20 @@
 #' demo_continuous(c(1, 1000), label = label_number_si(unit = "m"))
 #'
 #' demo_log10(c(1, 1e9), breaks = log_breaks(10), labels = label_number_si())
+#' demo_log10(c(1e-9, 1), breaks = log_breaks(10), labels = label_number_si(unit = "m"))
 label_number_si <- function(accuracy = 1, unit = NULL, sep = NULL, ...) {
```

Comment:
Since this PR has already been labeled as a breaking change, would it be ok to also change some default values?

- Could we also elevate `unit` to a required argument? I don't think there is ever a good reason to use `label_number_si()` without a unit, but users can always input an empty string if needed.
- `accuracy = NULL` to automatically choose precision based on data.
``` r
library(scales)

x <- c(1000, 1500, 2000)
label_number_si(accuracy = 1)(x)
#> [1] "1k" "2k" "2k"
label_number_si(accuracy = NULL)(x)
#> [1] "1.00k" "1.50k" "2.00k"
```

<sup>Created on 2021-03-17 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

Reply:
Unitless measure such as counts and ratio seem to me a reasonable use of `label_number_si`. I don't see a strong benefit to making `unit` mandatory.

> `accuracy = NULL` to automatically choose precision based on data.

Could you elaborate? What would the default `accuracy` be?

```
label_number_si(accuracy = NULL)(x)
#> [1] "1.00k" "1.50k" "2.00k"
```

In this example, are you suggesting that it could default to `accuracy=1` rather than the current `accuracy=3`? That seems useful to me.

Reply:
The default value of `accuracy` is currently `1`, but I'm suggesting to change it to `NULL`. This argument is simply passed to the `scales::number()` function. When `accuracy = NULL`, `scales::number()` automatically chooses the precision to display.

The example shows how using `accuracy = NULL` is advantageous. When `accuracy = 1` (the current default), the 2nd and 3rd elements of the output vector are identical.

Reply:
Thank you for the explanation! Yes, `accuracy = NULL` indeed seems like a better default than `accuracy = 1`.